sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala

```diff
@@ -451,3 +451,21 @@ object MetadataAttribute {
     } else None
   }
 }
+
+/**
+ * The internal representation of the FileSourceMetadataAttribute,
+ * it sets `__metadata_col` to `true` in AttributeReference metadata with the attr name `_metadata`
+ * - unapply() will check if an attribute reference is the file source metadata attribute reference
+ */
+object FileSourceMetadataAttribute {
+
+  val ATTRIBUTE_NAME = "_metadata"
```

Comment:
After more thought, I think it's more robust to not rely on the attribute name to extract file source metadata attrs. We can add a new metadata key to indicate file source metadata attributes. Sorry for the back and forth!

Reply:
np! sounds good to me!